recipe/meta.yaml

```diff
@@ -12,26 +12,34 @@ source:
 build:
   noarch: python
   number: 0
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
   host:
     - pip
     - poetry-core >=1.0.0
```

Comment:
I believe we're still using exact version pinnings in the host section

Reply:
That check passes in the linter so ¯\_(ツ)_/¯

Reply:
Yeah I wasn't aware that change was made

Reply:
The recipe standard says the following:

> Version pinnings must have exact versions; comparison operators must not be used. The version number should be the lowest possible. Version pinnings may be skipped for python build packages and packages pinned inside a conda_build_config.yaml file.

`poetry-core` is such an exception because it is a build tool like `pip`, `setuptools`, etc.

There is definitely some refinement needed about which packages need to be pinned and which don't, even though it's not bad being more conservative.

Reply:
How can a noob like me better discern a build tool from a standard package?

Reply:
That's unfortunately something you just have to read up about. It's not something you can discern by name. To make matters worse, the ecosystem is getting more fractured.

You can check the current linter exceptions here:

https://github.com/anaconda-distribution/anaconda-linter/blob/main/anaconda_linter/lint/check_build_help.py#L32-L51

and here:

https://github.com/anaconda-distribution/anaconda-linter/blob/main/anaconda_linter/lint/check_build_help.py#L119-L123

This is probably not exhaustive. And overall, I'm not sure if there is ever an exhaustive rule that goes beyond "if it's possible for a package update to break an existing build, pin the version".